src/webgl/p5.RendererGL.Retained.js

```diff
@@ -50,32 +69,41 @@ p5.RendererGL.prototype.createBuffers = function(gId, obj) {
     shader = this.setShader(this._getColorShader());
   }
 
-  // allocate space for vertex positions
-  var data = new Float32Array(vToNArray(obj.vertices));
+  /****THIS IS HOW VERTICES WERE DRAWN BEFORE****/
+
+  // // allocate space for vertex positions
+  // var data = new Float32Array(vToNArray(obj.vertices));
+  // shader.enableAttrib(shader.attributes.aPosition.location,
+  //   3, gl.FLOAT, false, 0, 0);
+  // gl.bindBuffer(gl.ARRAY_BUFFER, this.gHash[gId].vertexBuffer);
+  // gl.bufferData(gl.ARRAY_BUFFER, data, gl.STATIC_DRAW);
+
+  var data = new Float32Array(flatten(flatten(obj.lineVertices)));
   shader.enableAttrib(shader.attributes.aPosition.location,
     3, gl.FLOAT, false, 0, 0);
-  gl.bindBuffer(gl.ARRAY_BUFFER, this.gHash[gId].vertexBuffer);
+  gl.bindBuffer(gl.ARRAY_BUFFER, this.gHash[gId].lineVertexBuffer);
   gl.bufferData(gl.ARRAY_BUFFER, data, gl.STATIC_DRAW);
```

Comment:
I definitely think it would be possible to abstract it to that point. If a generic function was handling the buffer prepping it may have to take an unseemly number of arguments in order to be specific enough for all of the cases but I don't think that is really much of an issue and I think it would further the approach that the shader and texture class have been taking with which is a good approach imo. All of the work in that direction seems to really help with legibility.

This sounds fun and feasible. Would you mind me swiping the task?